core/tests/DesktopTests/AdalHeadlessAutomationTestApp/App/AuthenticationHelper.cs

```diff
@@ -0,0 +1,375 @@
+                return output.FromDictionaryToJson();
+                             
+            });
+
+            return await myTask.ConfigureAwait(false);
+        }
+
+        public static async Task<string> InvalidateRefreshToken(Dictionary<string, string> input)
+        {
+            Dictionary<string, object> output = new Dictionary<string, object>();
+            Task<string> myTask = Task<string>.Factory.StartNew(() =>
+            {
+                try
+                {
+                    TokenCache.DefaultShared.ReadItems();
+                    List<KeyValuePair<TokenCacheKey, AuthenticationResultEx>> CacheItems = QueryCache(input["authority"],
+                    input["client_id"], input["user_identifier"]);
+
+                    foreach (KeyValuePair<TokenCacheKey, AuthenticationResultEx> item in CacheItems)
+                    {
+                        var updated = item;
+                        updated.Value.RefreshToken = "bad_refresh_token";
+                        updated.Value.Result.ExpiresOn = DateTime.UtcNow;
+                        UpdateCache(item, updated);
+                    }
+                    //Send back error if userId or displayableId is not sent back to the user
+                    output.Add("invalidated_refresh_token_count", CacheItems.Count.ToString());                   
+                }
+                catch (Exception exc)
+                {
+                    output.Add("error", exc.Message); 
+                }
+                return output.FromDictionaryToJson();
+            });
+
+            return await myTask.ConfigureAwait(false);
+        }
+
+        public static async Task<string> ReadCache()
+        {
+            Task<string> myTask = Task<string>.Factory.StartNew(() =>
+            {
+
+                Dictionary<string, object> output = new Dictionary<string, object>();
+                TokenCache.DefaultShared.ReadItems();
+                var list = TokenCache.DefaultShared.tokenCacheDictionary;
+                
+                if (list.Any())
+                {
+                    output.Add("Count", list.Count());
+                    // TODO: only returns the first item in the cache. Why?
+                    // This causes problems with the headless automation tests as there will be
+                    // entries in the cache for the test account the test is running as (authentication
+                    // is required to access the passwords in KeyVault).s
+                    var item = list.FirstOrDefault();
+                    output.Add("AccessToken", item.Value.Result.AccessToken);
+                    output.Add("expires_on", item.Value.Result.ExpiresOn);
+                    output.Add("refresh_token", item.Value.RefreshToken);
+                }
+              
+                return FromDictionaryToJson(output);
+            });
+            return await myTask.ConfigureAwait(false);
```

Comment:
It's easier to do Task.FromResult() to transform a sync method into an async one